tmail-backend/tmail-third-party/openpaas/src/test/java/com/linagora/tmail/contact/OpenPaasContactsConsumerTest.java

```diff
@@ -0,0 +1,133 @@
+package com.linagora.tmail.contact;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.james.backends.rabbitmq.Constants.EMPTY_ROUTING_KEY;
+import static org.apache.james.backends.rabbitmq.RabbitMQExtension.IsolationPolicy.WEAK;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.awaitility.Durations.TEN_SECONDS;
+
+import org.apache.james.backends.rabbitmq.RabbitMQExtension;
+import org.apache.james.jmap.api.model.AccountId;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.rabbitmq.OutboundMessage;
+
+import com.linagora.tmail.OpenPaasConfiguration;
+import com.linagora.tmail.api.OpenPaasRestClient;
+import com.linagora.tmail.api.OpenPaasServerExtension;
+import com.linagora.tmail.james.jmap.contact.EmailAddressContactSearchEngine;
+import com.linagora.tmail.james.jmap.contact.InMemoryEmailAddressContactSearchEngine;
+
+class OpenPaasContactsConsumerTest {
+
+    @RegisterExtension
+    static OpenPaasServerExtension openPaasServerExtension = new OpenPaasServerExtension();
+
+    @RegisterExtension
+    static RabbitMQExtension rabbitMQExtension = RabbitMQExtension.singletonRabbitMQ()
+        .isolationPolicy(WEAK);
+
+    private EmailAddressContactSearchEngine searchEngine;
+    private OpenPaasContactsConsumer consumer;
+
+    @BeforeEach
+    void setup() throws URISyntaxException {
+        OpenPaasRestClient restClient = new OpenPaasRestClient(
+            new OpenPaasConfiguration(
+                openPaasServerExtension.getBaseUrl(),
+                "admin",
+                "admin")
+        );
+        searchEngine = new InMemoryEmailAddressContactSearchEngine();
+        consumer = new OpenPaasContactsConsumer(rabbitMQExtension.getReceiverProvider(),
+            rabbitMQExtension.getSender(),
+            rabbitMQExtension.getRabbitMQ().withQuorumQueueConfiguration(),
+            searchEngine, restClient);
+
+        consumer.start();
+    }
+
+    @AfterEach
+    void afterEach() throws IOException {
+        consumer.close();
+    }
+
+    @Test
+    void consumeMessageShouldNotCrashOnInvalidMessages() throws InterruptedException {
```

Comment:
Testcase is failing

Reply:
Maybe we can paste here the error so that reviewers do not need to run the test themselves in order to figure out what is going on?

Reply:
It's fixed now.